packages/hoc-disabled-tooltip/src/disabledTooltip.js

```diff
@@ -19,50 +19,30 @@ function disabledTooltip(wrappedComponentType) {
   const WrappedComponent = props =>
     React.createElement(wrappedComponentType, props);
   return class extends React.Component {
-    constructor(props) {
-      super(props);
-      this.state = {
-        showTooltip: false
-      };
-      this.onMouseOver = this.onMouseOver.bind(this);
-      this.onMouseLeave = this.onMouseLeave.bind(this);
-    }
-
-    onMouseOver(event) {
-      this.setState({ showTooltip: !this.props.noTooltip });
-      return this.props.onMouseOver(event);
-    }
-
-    onMouseLeave(event) {
-      this.setState({ showTooltip: false });
-      return this.props.onMouseLeave(event);
-    }
-
     render() {
+      const WrappedComponentWithTooltip = withTooltip(WrappedComponent);
+
       const {
         displayBlock,
         noTooltip,
         tooltipAlign,
         tooltipText,
+        onMouseOver,
+        onMouseLeave,
         ...wrappedComponentProps
       } = this.props;
+
       const wrapperProps = { displayBlock };
-      const hitAreaProps = {
-        onMouseOver: this.onMouseOver,
-        onMouseLeave: this.onMouseLeave
-      };
+
       return (
         <Wrapper {...wrapperProps}>
-          <div className="hitWrapper">
-            <WrappedComponent {...wrappedComponentProps} />
-            {this.props.disabled && <div className="hit" {...hitAreaProps} />}
-          </div>
-          {this.props.disabled &&
-            this.state.showTooltip && (
-              <div>
-                <Tooltip text={tooltipText} align={tooltipAlign} />
-              </div>
-            )}
+          <WrappedComponentWithTooltip
+            onMouseOver={onMouseOver}
+            onMouseLeave={onMouseLeave}
+            disableTooltip={!this.props.disabled || noTooltip}
+            tooltipText={tooltipText}
+            {...wrappedComponentProps}
```

Comment:
`onMouseOver`, ` onMouseLeave`, `tooltipAlign` and `tooltipText` could be part of the `wrappedComponentProps`. No need to extract from props and then passed back to the wrapped component with the same names.